instrumentation/trilogy/lib/opentelemetry/instrumentation/trilogy.rb

```diff
@@ -11,6 +11,45 @@ module OpenTelemetry
   module Instrumentation
     # Contains the OpenTelemetry instrumentation for the Trilogy gem
     module Trilogy
+      extend self
+
+      CURRENT_ATTRIBUTES_KEY = Context.create_key('mysql-attributes-hash')
```

Comment:
Although these two context keys will have a different has entry I think it will be confusing to me if I inspected in the has in a debugger and saw that they key name was `mysql-attributes-hash`. 

I also wonder if you all will be running the mysql driver and trilogy drivers simultaneously in your apps. 

Would having a common `MySQL::Context` helper make sense?

Reply:
Applications should be using either Mysql2 or Trilogy as the database client, but not both simultaneously (am I understanding your question correctly?). Either way, having a common MySQL::Context SGTM 👍 Ultimately they're both propagating context for a SQL database. I can go ahead and extract a helper.

Can you clarify whether you think the context key should change? Maybe we should name them `mysql2-attributes-hash` and `trilogy-attributes-hash` respectively?

Reply:
> Applications should be using either Mysql2 or Trilogy as the database client, but not both simultaneously (am I understanding your question correctly?). Either way, having a common MySQL::Context SGTM 👍 Ultimately they're both propagating context for a SQL database. I can go ahead and extract a helper.

My question was specifically about your migration process to Trilogy. I imagine your more complex applications are the like the GH monolith where we have may have multiple databases so I do imagine a world where you may have both drivers and instrumentations running in parallel. 

> Can you clarify whether you think the context key should change? Maybe we should name them mysql2-attributes-hash and trilogy-attributes-hash respectively? Or would we pull everything out to the shared SQL helper, including the key?

If it is a shared helper then the key should be the same. If you stick with the Trilogy specific Context key then I think naming it `trilogy-attributes-hash` would be best.

Reply:
> My question was specifically about your migration process to Trilogy.

Gotcha. At Shopify, we intend to only be using a single db client per application, even in our multi-db applications. Migrating to Trilogy will involve migrating all of our dbs at the same time, so on our end I don't think there's a concern with sharing a key between the Mysql2 / Trilogy drivers.

I'll stick with the shared helper and a single key for now.

Reply:
Thank you!

Reply:
Hey @arielvalentin I think we should probably pull the mysql context into common similarly to how we did it with the HTTP context. We could duplicate the behaviour of mysql into trilogy for now, and then extract it after. WDYT?

Reply:
> Hey @arielvalentin I think we should probably pull the mysql context into common similarly to how we did it with the HTTP context. We could duplicate the behaviour of mysql into trilogy for now, and then extract it after. WDYT?

Works for me. Thank you friends!